components/hook-cjs/void/index.mjs

```diff
@@ -1,9 +1,19 @@
 const { URL } = globalThis;
+
 const { search: __search } = new URL(import.meta.url);
 
-const { expect } = await import(`../../expect/index.mjs${__search}`);
-export const { noop: unhook } = await import(`../../util/index.mjs${__search}`);
+const { ExternalAppmapError } = await import(
+  `../../error/index.mjs${__search}`
+);
+const { logErrorWhen } = await import(`../../log/index.mjs${__search}`);
+const { assert, noop } = await import(`../../util/index.mjs${__search}`);
+
+export const unhook = noop;
 
 export const hook = (_agent, { hooks: { cjs } }) => {
-  expect(!cjs, "expected configuration to disable cjs module hook");
+  assert(
+    !logErrorWhen(cjs, "No support for recordering common-js modules"),
```

Comment:
You got the condition in the log reversed. Here is the correct branching:
- If cjs is `truthy`, an error is *logged* and the assertion fails.
- If cjs is `falsy`, an error is *not logged* and the assertion passes.

Reply:
We could align the condition of the log on the assertion. But I don't think it is more readable:

```js
assert(
  logErrorWhenNot(!cjs, "No support for recording common-js modules"),
  "Expected hooks.cjs to be false",
  ExternalAppmapError,
);
```

Reply:
https://github.com/getappmap/appmap-agent-js/pull/138/files#diff-da1eca588b0fb0bee76240a2303f134eabe0901c16680407571783e4bc7165b5R114-R115
```js
logError("Could not recognize %j as a mocha command", tokens);
throw new ExternalAppmapError("Not a parsed mocha command");
```
It may just be me, but this is a lot more digestible as it's intent is clearly stated

Reply:
I agree, but they don't have a condition. So it would have to be:
```js
if (cjs) {
  logError("No support for recording common-js modules");
  throw new ExternalAppmapError("Expected hooks.cjs to be false");
}
```

Which is admittedly more readable. But it adds more branching to the code and I try to reach 100% coverage. Is this stupid? I don't want to add `/* c8 ignore */` around every assertion... Sometimes testability harms the code I guess?